GliaWidgets/CallVisualizer/CallVisualizer.swift

```diff
@@ -0,0 +1,51 @@
+import Foundation
+import UIKit
+import SalemoveSDK
+
+public final class CallVisualizer {
+
+    private let coreSdk: CoreSdkClient
+
+    init(coreSdk: CoreSdkClient) {
+        self.coreSdk = coreSdk
+    }
+
+    private func presentOnTopViewController(viewController: UIViewController) {
+        if let rootViewController = UIApplication.shared.keyWindow?.rootViewController {
+            rootViewController.present(viewController, animated: false)
+        }
+    }
+
+    public func requestVisitorCode(completion: @escaping (Result<VisitorCode, Error>) -> Void) {
+        _ = coreSdk.requestVisitorCode(completion)
+    }
+
+    public func requestVisitorCodeViewController(uiConfig: RemoteConfiguration? = nil) -> UIViewController {
+        let vc = VisitorCodeViewController(viewType: .alert, uiConfig: uiConfig)
+        vc.modalPresentationStyle = .overFullScreen
+        vc.modalTransitionStyle = .crossDissolve
+        return vc
+    }
+
+    public func presentVisitorCodeViewController(uiConfig: RemoteConfiguration? = nil) {
```

Comment:
Might be because I lack experience with iOS but why 'layout' and not something like 'show'? Is it somehow related to how other methods are named or this work better represents the logic involved?

Reply:
irrelevant in new data-driven approach